libs/transactions/helpers/units.ts

```diff
@@ -0,0 +1,105 @@
+import { toBN } from "web3-utils";
+import type BN from "bn.js";
+
+/**
+ * Validates that given value is string
+ *
+ * @function isString
+ * @param {unknown} value transaction value
+ * @returns {boolean}
+ */
+export function isString(s: unknown): boolean {
+  return typeof s === "string" || s instanceof String;
+}
+
+/**
+ * Validates that given value string does not start with `-` ie negative
+ *
+ * @function isPositive
+ * @param {string} value transaction value
+ * @returns {boolean}
+ */
+export function isPositive(value: string): boolean {
+  return value.substring(0, 1) !== "-";
+}
+
+/**
+ * Validates that given value is not just `.`
+ *
+ * @function isValueValid
+ * @param {string} value transaction value
+ * @returns {boolean}
+ */
+export function isValueValid(value: string): boolean {
+  return value !== ".";
+}
+
+/**
+ * Validates that splitting value returns 2 items max (ie whole and fraction)
+ *
+ * @function isDecimalPartValid
+ * @param {string} value transaction value
+ * @returns {boolean}
+ */
+export function isDecimalPartValid(value: string): boolean {
+  const parts = value.split(".");
```

Comment:
If the value is undefined we get an error.

Reply:
TBD validations && toBaseUnit